### PR TITLE
TPSVC-14487: Negative events not capture when manually generating audit logs

### PR DESCRIPTION
**Pull request by jehervy**
**What is the problem this Pull Request is trying to solve?**
In some cases (login flow), HTTP requests return `3xx REDIRECTION` codes. These responses must generate audit logs as well as `2xx SUCCESSFUL` codes

**What is the chosen solution to this problem?**

Consider `3xx REDIRECTION` http code as successful and generate corresponding audit logs in the Aspect.

**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorInterceptor.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public void afterCompletion(HttpServletRequest request, HttpServletResponse resp
                     .map(this::extractRequestType).map(t -> this.parse(requestBodyString, t)).orElse(null);
             String responseBodyString = Optional.ofNullable(response).map(this::extractContent).orElse(null);
             // Only log if code is not successful
-            if (!HttpStatus.valueOf(responseCode).is2xxSuccessful()) {
+            if (!HttpStatus.valueOf(responseCode).is2xxSuccessful() && !HttpStatus.valueOf(responseCode).is3xxRedirection()) {
```
Comment (fhuaulme) [inline daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorInterceptor.java:66]: Would it be simpler to write `HttpStatus.valueOf(responseCode).isError()` ?

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java [hunk 2]
```diff
@@ -71,7 +71,8 @@ public Object auditLogGeneration(final ProceedingJoinPoint proceedingJoinPoint)
             }
 
             // Send logs only in case of success or if responseCode can't be defined
-            if (responseCode == 0 || HttpStatus.valueOf(responseCode).is2xxSuccessful()) {
+            if (responseCode == 0 || HttpStatus.valueOf(responseCode).is2xxSuccessful()
```
Comment (fhuaulme) [inline daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java:74]: Would it be simpler to write `!HttpStatus.valueOf(responseCode).isError()` ?

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorInterceptor.java (hunk 1 above)
Comment (jehervy) [reply daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorInterceptor.java]: Much simpler indeed 🤦 Thanks for feedback !

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java (hunk 2 above)
Comment (jehervy) [reply daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java]: Completely 👌

**Pull request merged**